### ISSUE TITLE
Can't shutdown client if used configure method

### ISSUE THREAD

**Issue by pankajmi** [dispatch/reboot#59]
I am using dispatch_core version 0.10.1 

Http client created using configure method doesn't shutdown if we don't set excecutors explicitly. Is it expected ?

val http = new Http()
      .configure(builder => {
        builder.setConnectionTimeoutInMs(1000)
        //builder.setExecutorService(Executors.newCachedThreadPool())
        builder
      })
http.shutdown

you can have a look at part of jstack - 
https://gist.github.com/pankajmi/5812305

EDIT: and CPU usage is very very high

**Comment by davidshen84** [comment]
Do we have any formal decision on how to resolve this issue? I am hit by this problem as well. I think the problem is caused by the `Http` companion object. When you do:

```scala
dispatch.Http.configure
```

or

```scala
new dispatch.Http().configure(...)
```

The `dispatch.InternalDefaults.client` function was evaluated and a client instance was created. One way or the other, you have to shutdown this hidden client so your application can exit.

Currently, if I need a customized client, my walk around is to avoid the default client got created.

````scala
  val http = new Http(
    new AsyncHttpClient(
      new AsyncHttpClientConfig.Builder()
        .setFollowRedirect(true)
        .build()
    )
  )
```

I provide the `Http` constructor a customized client, so it will not evaluate the default client function. And `http.shutdown()` will shutdown the only client and let my application exit properly. But I do hope we can use `Http.configure` to build customized client.

I am thinking maybe we can put the `configure` definition in the `Http` object, instead of the class.

**Comment by farmdawgnation** [comment]
This has been resolved in Dispatch 0.12.x and Dispatch 0.13.x, and releases including this fix are forthcoming.

For API compatibility reasons, I didn't change the current behavior in 0.12, but I did cause it to start throwing deprecation warnings when it is used and provide Http.closeAndConfigure to implement the new, desired behavior.
